api/src/main/java/org/apache/gravitino/tag/SupportsTags.java

```diff
@@ -58,6 +59,8 @@ public interface SupportsTags {
    * @param tagsToAdd The arrays of tag name to be added to the object.
    * @param tagsToRemove The array of tag name to be removed from the object.
    * @return The array of tag names that are associated with the object.
+   * @throws TagAlreadyAssociatedException If the tag is already associated with the object.
    */
-  String[] associateTags(String[] tagsToAdd, String[] tagsToRemove);
+  String[] associateTags(String[] tagsToAdd, String[] tagsToRemove)
```

Comment:
What will happen if we associate non-existed tags?

Reply:
Nothing will be happened, it is just an no-op.

Reply:
If so, suggest adding comments to the API. Otherwise, some users may think this is an asynchronous operation that allows users to bind tags before creating them.

Reply:
OK, I will fix this in #4257

Reply:
I fixed in #4257